chat-widget/src/plugins/createChatTranscript.ts

```diff
@@ -0,0 +1,395 @@
+                                });
+                            }
+
+                            observers.push(observer);
+
+                            return () => {
+                                observers = observers.filter(o => o !== observer);
+
+                                if (!observers.length) {
+                                    subscription.unsubscribe();
+                                    subscription = null;
+                                }
+                            };
+                        });
+                    }
+                <\/script>
+                <script>
+                    const messages = ${JSON.stringify(this.options.messages)};
+                <\/script>
+                <script>
+                    class Translator {
+                        static convertTranscriptMessageToActivity(message) {
+                            const {created, isControlMessage, content, tags, from, attachments, amsMetadata, amsReferences} = message;
+                            const activity = {
+                                from: {
+                                    role: 'bot'
+                                },
+                                type: 'message'
+                            };
+
+                            // Ignore control messages
+                            if (isControlMessage) {
+                                return false;
+                            }
+
+                            if (tags) {
+                                const formattedTags = tags.split(',');
+
+                                // Ignore system message
+                                if (formattedTags.includes('system')) {
+                                    return false;
+                                }
+                            }
+
+                            // Add C1 user display name
+                            if (message.from.user && message.from.user.displayName) {
+                                activity.from.name = message.from.user.displayName;
+                            }
+
+                            // Add C2 user display name
+                            if (from && from.application && from.application.displayName && from.application.displayName === 'Customer') {
+                                activity.from = {
+                                    role: 'user',
+                                    name: from.application.displayName
+                                };
+                            }
+
+                            // Attachments
+                            if (amsReferences && amsMetadata) {
+                                const metadata = JSON.parse(amsMetadata);
+                                const { fileName } = metadata[0];
+                                const text = \`The following attachment was uploaded during the conversation: \${fileName}\`;
+
+                                if (message.attachments && message.contentUrl) {

```

Comment:
Is this code being hit? When sending attachments, I see in the downloaded transcript just the text 
<img width="301" alt="image" src="https://user-images.githubusercontent.com/13504205/235548297-f9db47fd-9c31-4b0c-9eaa-1175d31b7de6.png">

Reply:
Not for the moment. It will be needed if we decided to support attachment rendering. Right now we're only display a message that an attachment has be uploaded.